### PR TITLE
lexicon: 3.9.4 -> 3.11.2

### PR DESCRIPTION
**Pull request by blitz**
###### Description of changes

This PR unbreaks lexicon and at the same time also updates it to 3.11.2.

I've switched to poetry2nix for packaging, because lexicon demands some older versions of python libraries (e.g. click). With the move to poetry2nix, we can also remove some attributes in `pythonPackages` that were only imported for lexicon to begin with.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@flyfloh

### CODE REVIEW

File: pkgs/tools/admin/lexicon/default.nix [hunk 1]
```diff
@@ -1,60 +1,24 @@
...
-  src = fetchFromGitHub {
+  python = python3;
+
+  projectDir = fetchFromGitHub {
     owner = "AnalogJ";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-TySgIxBEl2RolndAkEN4vCIDKaI48vrh2ocd+CTn7Ow=";
+    repo = "lexicon";
+    rev = "v3.11.2";
```
Comment (tfc) [inline pkgs/tools/admin/lexicon/default.nix:18]: this version information is also available in the toml file. does it make sense to get it from there? this way it also becomes impossible to accidentally check out a rev that does not match the toml file's description.

Comment (blitz) [reply pkgs/tools/admin/lexicon/default.nix]: Fixed. It's not necessarily beautiful, though. :)

**Comment by SuperSandro2000** [comment]
poetry2nix is highly discouraged as it can pin vurnable dependencies and cause weird eval errors when dependencies are no longer compatible instead of build failures. Also adds it more files into nixpkgs which bloats the repository size.

If you require other versions of dependency the Python Team highly recommends to use [package overrides](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-override-a-python-package-how-to-override-a-python-package).

File: pkgs/tools/admin/lexicon/default.nix [hunk 2]
```diff
@@ -1,60 +1,27 @@
...
-  ];
-
-  disabledTestPaths = [
-    # Tests require network access
-    "lexicon/tests/providers/test_auto.py"
-  ];
-
-  pythonImportsCheck = [
-    "lexicon"
-  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/lexicon/default.nix:56]: Removing testing is also a really bad idea and generally causes a package to rot over time and silently break.

File: pkgs/tools/admin/lexicon/poetry.lock [hunk 3]
```diff
@@ -0,0 +1,1741 @@
...
+
+[package.dependencies]
+tomli = {version = "*", optional = true, markers = "extra == \"toml\""}
+
+[package.extras]
+toml = ["tomli"]
+
+[[package]]
+name = "cryptography"
+version = "3.4.7"
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/lexicon/poetry.lock:175]: That version is horrible out of date.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by blitz** [comment]
@SuperSandro2000 Thanks for the input. I'll discuss this with @flyfloh and retry. :)

**Comment by blitz** [comment]
> @SuperSandro2000 Thanks for the input. I'll discuss this with @flyfloh and retry. :)

There doesn't seem to be a good way to fix lexicon besides sticking to their poetry file. The tests don't succeed. Given that lexicon is broken right now and few people except @flyfloh and me seem to care, I'm also happy with removing it in nixpkgs.

**Comment by dotlambda** [comment]
> There doesn't seem to be a good way to fix lexicon besides sticking to their poetry file.

Actually, it's rather easy to fix using our conventional `pythonPackages`: https://github.com/NixOS/nixpkgs/pull/176287

**Comment by blitz** [comment]
> > There doesn't seem to be a good way to fix lexicon besides sticking to their poetry file.
> 
> Actually, it's rather easy to fix using our conventional `pythonPackages`: #176287

Indeed. I gave up when there were test failures all around. ;) Thanks for picking this up!

**Pull request closed without merging**